paddle/fluid/framework/details/op_handle_base.cc

```diff
@@ -60,6 +60,7 @@ void OpHandleBase::RecordWaitEventOnCtx(platform::DeviceContext *waited_ctx) {
 #ifdef PADDLE_WITH_CUDA
   if (platform::is_cpu_place(waited_ctx->GetPlace()) || events_.empty()) {
     for (auto &dev_ctx : dev_ctxes_) {
+      PADDLE_ENFORCE_NOT_NULL(dev_ctx.second);
```

Comment:
Please add a check in line67 too.

Reply:
Check whether the `waited_ctx` is empty? I think adding it before line 61 is better.